data/lua/wml-tags.lua

```diff
@@ -365,6 +365,16 @@ local function handle_event_commands(cfg)
 			end
 			arg = wesnoth.tovconfig(arg)
 		end
+		-- Interruption tags need special treatment, so they're hardcoded here
+		if cmd == "continue_if" then
+			if context_type ~= "loop" then
+				helper.wml_error("[continue] appeared outside of a loop", cmd)
+			elseif wesnoth.eval_conditional(arg) then
+				return "continue"
+			end
+		elseif cmd == "break_if" and wesnoth.eval_conditional(arg) then
```

Comment:
That was intentional, actually - it means "break" can double as "return". I could disable it in `[if]` or `[switch]`, I suppose, but I probably can't disable it in a nested `[command]` block.